### PR TITLE
[dask] Course rationale

### PR DESCRIPTION
**Pull request by dkillick**
Here is some draft content on the rationale for this course: why it exists and what it intends to teach. It is very much still a **draft** however -- there may even still be typos -- so go easy on the review!

### CODE REVIEW

File: course_content/notebooks/dask_intro.ipynb [hunk 1]
```diff
@@ -4,9 +4,45 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Course overview\n",
+    "\n",
+    "Dask is a flexible parallel computing library for Python. It allows dynamic scheduling of tasks for optimising computations and includes collections (extensions of arrays, lists and dataframes) for use with parallel data processing.\n",
+    "\n",
+    "As of Iris v2.0, dask has been integrated with Iris. This means you can utilise all of the benefits of dask while keeping the fileformat interoperability and other functionality that Iris provides."
```
Comment (bjlittle) [inline course_content/notebooks/dask_intro.ipynb]: @dkillick Minor ... but are we using `dask` or `Dask` ... there is a mixture of capitalization used throughout this document.

File: course_content/notebooks/dask_intro.ipynb [hunk 2]
```diff
@@ -4,9 +4,45 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Why does this course exist?\n",
+    "\n",
+    "This course exists to enable you to use dask and Iris together for more performant, memory efficient data processing by showing useful patterns for combining dask and Iris.\n",
```
Comment (bjlittle) [inline course_content/notebooks/dask_intro.ipynb]: ... `by showing SOME useful patterns` ...

File: course_content/notebooks/dask_intro.ipynb [hunk 3]
```diff
@@ -4,9 +4,45 @@
...
+    "\n",
+    "#### Iris/dask integration\n",
+    "\n",
+    "Iris uses dask to defer data loading and processing by default in a number of situations, including loading and some mathematical, analytical and statistical operations. This course will introduce patterns for producing your own bespoke data processing pipelines that leverages Iris and dask functionality \n",
+    "\n",
+    "This course will introduce some patterns for using Iris and dask together to produce your own bespoke data prcessing pipelines that meet your analytical needs while maintaining deferred data.\n",
```
Comment (bjlittle) [inline course_content/notebooks/dask_intro.ipynb]: This sentence is almost exactly the same as the previous one ... is that meant?

File: course_content/notebooks/dask_intro.ipynb [hunk 4]
```diff
@@ -4,9 +4,45 @@
...
+    "\n",
+    "\n",
+    "#### Performance\n",
+    "\n",
+    "A real benefit of dask is parallel data processing by default. This means you will see existing Iris processing paradigms getting faster, and we will show patterns for loading and processing data to take advantage of dask parallel processing so that your bespoke processing pipelines will also run faster.\n",
+    "\n",
+    "\n",
+    "#### Memory usage\n",
+    "\n",
+    "Data volumes are getting larger and larger, and we are getting to the point where a typical dataset is now larger than the amount of system memory available to process the dataset with. In many cases then, we can no longer just load an entire dataset into memory and perform data processing on the dataset.\n",
```
Comment (bjlittle) [inline course_content/notebooks/dask_intro.ipynb]: I feel that users should be exposed to the notion of `out-of-core processing` as a concept along with deferred data and lazy evaluation. Can that be blended in here? Using the terms without over explaining them

File: course_content/notebooks/dask_intro.ipynb (hunk 2 above)
Comment (dkillick) [reply course_content/notebooks/dask_intro.ipynb]: Nono, we're showing *all* the useful patterns 😜 

Haha, I'll make the change.

File: course_content/notebooks/dask_intro.ipynb (hunk 1 above)
Comment (dkillick) [reply course_content/notebooks/dask_intro.ipynb]: We should be using however dask references itself. However, I've never managed to figure out what that is, so I've never been able to choose one to enforce!

File: course_content/notebooks/dask_intro.ipynb (hunk 3 above)
Comment (dkillick) [reply course_content/notebooks/dask_intro.ipynb]: Hm, looks like one of them got missed from being removed as I was writing this all - hence this still being a draft!

File: course_content/notebooks/dask_intro.ipynb (hunk 4 above)
Comment (dkillick) [reply course_content/notebooks/dask_intro.ipynb]: Yup, good point 👍

**Comment by pp-mo** [comment]
Handy link ...
https://nbviewer.jupyter.org/github/dkillick/courses/blob/dask_rationale/course_content/notebooks/dask_intro.ipynb

File: course_content/notebooks/dask_intro.ipynb (hunk 1 above)
Comment (dkillick) [reply course_content/notebooks/dask_intro.ipynb]: Right, I decided on "dask". I've implemented that decision throughout the notebook.

**Comment by pp-mo** [comment]
Nice tight addition :ok_hand: 
Any further improvements are probably additions -- but may be better off without.

**Comment by dkillick** [comment]
Thanks @pp-mo!

**Pull request merged**